src/main/java/TaskList.java

```diff
@@ -1,37 +1,41 @@
+import java.util.ArrayList;
+
 public class TaskList {
-    private Task[] list;
-    private int listPointer = 0;
+    private ArrayList<Task> list;
 
-    public TaskList(int size) {
-        list = new Task[size];
+    public TaskList() {
+        list = new ArrayList<Task>();
     }
 
     public void addTask(Task task) {
-        list[listPointer] = task;
-        listPointer++;
+        list.add(task);
+    }
+
+    public Task deleteTask(int index) {
```

Comment:
The addition of the `deleteTask` functionality is a great enhancement. I noticed that you're checking if the input equals `"delete"` for handling the absence of an index. Have you considered providing more user-friendly feedback when an index is not provided? For example, a message prompting the user to specify which task to delete might help improve the user experience.